src/recommendationservice_ballerina/src/recommendationservice/RecommendationService_sample_service.bal

```diff
@@ -1,13 +1,70 @@
 import ballerina/grpc;
+import ballerina/io;
+import ballerina/kubernetes;
 
-listener grpc:Listener ep = new (9090);
+// Product catalog client.
+ProductCatalogServiceBlockingClient productCat = new ("http://productcatalogservice:3550");
 
-service RecommendationService on ep {
+@kubernetes:Service {
+    serviceType: "ClusterIP",
+    name: "recommendationservice"
+}
+service RecommendationService on new grpc:Listener(8080) {
 
     resource function ListRecommendations(grpc:Caller caller, ListRecommendationsRequest value) {
-        // Implementation goes here.
+        Empty req = {};
```

Comment:
Shall we define a module level `final Empty` variable to use for all requests? Never changes right?

Reply:
yeah will make this global and final